pkg/scheduler/framework/runtime/framework_test.go

```diff
@@ -757,12 +759,19 @@ func TestNewFrameworkMultiPointExpansion(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			fw, err := NewFramework(registry, &config.KubeSchedulerProfile{Plugins: tc.plugins})
-			if (err != nil && tc.wantErr == "") || (err == nil && tc.wantErr != "") || (err != nil && !strings.Contains(err.Error(), tc.wantErr)) {
-				t.Errorf("Unexpected error, got %v, expect: %s", err, tc.wantErr)
+			if err != nil {
+				if tc.wantErr == "" || !strings.Contains(err.Error(), tc.wantErr) {
+					t.Fatalf("Unexpected error, got %v, expect: %s", err, tc.wantErr)
+				}
+			} else {
+				if tc.wantErr != "" {
+					t.Fatalf("Unexpected error, got %v, expect: %s", err, tc.wantErr)
+				}
 			}
+
 			if tc.wantErr == "" {
```

Comment:
we can remove this `tc.wantErr` check now that it was expanded into the `else` statement above (in other words, we're already checking this)

Reply:
I don't think so, in this place, we will check plugins, not the error itself.

Reply:
Yes, we still need the `diff` check, but I'm referring to `if tc.wantErr == ""`, ie changing L772-776 from:
```
if tc.wantErr == "" {
  if diff := cmp.Dif(......)... {
    t.Err..
  }
}
```

to just the diff check,
```
  if diff := cmp.Dif(......)... {
    t.Err..
  }
```

Reply:
```
if err != nil && tc.wantErr != "" && strings.Contains(err.Error(), tc.wantErr) {
}
```
in this situation, we will not like to verify the plugins.